### PR TITLE
@types/braintree webhooks

### PR DESCRIPTION
**Pull request by ACDR**
Braintree webhook testing types

Please fill in this template.

- [Y] Use a meaningful title for the pull request. Include the name of the package modified.
- [Y] Test the change in your own code. (Compile and run.)
- [Y] Add or edit tests to reflect the change. (Run with `npm test`.)
- [Y] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [Y] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [Y] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [Y] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.braintreepayments.com/guides/webhooks/testing-go-live/node

- [N/A] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

- [N/A] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/braintree/index.d.ts [hunk 1]
```diff
@@ -816,6 +827,55 @@ declare namespace braintree {
...
+    /**
+     * Webhooks
+     */
+
+    export class SampleNotification {
+        bt_signature: string;
+        bt_payload: string;
+    }
+
+    export class WebhookNotification {
```
Comment (smrubin) [inline types/braintree/index.d.ts]: Looks like `timestamp` is part of the WebhookNotification as well, can we add it here? https://developers.braintreepayments.com/reference/general/webhooks/overview

File: types/braintree/index.d.ts [hunk 2]
```diff
@@ -816,6 +827,55 @@ declare namespace braintree {
...
+        kind: WebhookNotificationKind;
+        subscription?: Subscription;
+        merchantAccount?: MerchantAccount;
+        transaction?: Transaction;
+        dispute?: Dispute;
```
Comment (smrubin) [inline types/braintree/index.d.ts]: Looks like there are some additional kinds of webhook notifications as well that yield additional attributes (https://developers.braintreepayments.com/reference/general/webhooks/overview#notification-kinds). Think this PR is a great start to implementing webhooks though, so it shouldn't be a blocker. Feel free to submit another PR for those if you're up for it!

**Review by smrubin — CHANGES_REQUESTED** [review]
@ACDR Looks good and thanks for working on the webhooks (and not to mention adding types for the testing). I'm only requesting changes to add the `timestamp` field to the `WebhookNotification` class. After that change, should be good to ship.

**Review by smrubin — APPROVED** [review]
Thanks @ACDR!

**Comment by orta** [comment]
👍

**Pull request merged**